### ISSUE TITLE
[Truffle] Array#pack issue 

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#3547]
Expected

```
$ ruby -e 'puts ["metadata.gz", "0000444", "0000000", "0000000", "00000001244", "00000000044", "        ", " ", "0", nil, "ustar", "00", "wheel", "wheel", "0000000", "0000000", ""].pack("a100a8a8a8a12a12a7aaa100a6a2a32a32a8a8a155")'
metadata.gz0000444000000000000000000000124400000000044        0ustar00wheelwheel00000000000000
```

Actual

```
$ ./bin/jruby -X+T -e 'puts ["metadata.gz", "0000444", "0000000", "0000000", "00000001244", "00000000044", "        ", " ", "0", nil, "ustar", "00", "wheel", "wheel", "0000000", "0000000", ""].pack("a100a8a8a8a12a12a7aaa100a6a2a32a32a8a8a155")'
-e:1:in `pack': too few arguments (ArgumentError)
    from -e:1:in `<main>'
```

Found in rubygems.

**Comment by chrisseaton** [comment]
This is caused by a  bug in an optimisation where we attempt to recover loops in pack expressions so they can be vectorised. As of 64597b735e905e74eac40f96c5a9860c1b7e6323 you can workaround by doing `-Xtruffle.pack.recover=9999` and then it works.

**Comment by chrisseaton** [comment]
Fixed in 23a66696f659608c8289b6f0500b7e564dba993e.

**Comment by bjfish** [comment]
Re-opening this because it has regressed. The workaround still works: `-Xtruffle.pack.recover=9999`.

I've added a spec for this error here: 83ca4b6f3bd43c1e5c2638a4dece71b67728598d

**Comment by chrisseaton** [comment]
Fixed in f9a5f31fa8f38067082e375c7585f3482a4678fd
